packages/developer-extension/src/background/domain/logRequests.ts

```diff
@@ -0,0 +1,64 @@
+import { store } from '../store'
+
+const decoder = new TextDecoder('utf-8')
+chrome.webRequest.onBeforeRequest.addListener(
+  (info) => {
+    if (!store.logEvents) {
```

Comment:
It would be more convenient to log events as soon as they occur. In previous hackathon, we triggered events from sdk code, cf [branch](https://github.com/DataDog/browser-sdk/compare/hack-a-dog-chrome-extension#diff-c04d6d921a5dcb5a54e9ae53e0e79810a60da0ae380b288ee6526c8b75bf2b78R7).
It could be a future improvement though

Reply:
I like to see events when they are sent, from an external point of view, to observe what the SDK does instead of what the SDK tells us. Also printing events when they are generated could be misleading because not all events are sent (view updates could be discarded). This could be configured though.

Reply:
both can indeed be helpful